api/views.py

```
@@ -0,0 +1,148 @@
+
+            markdown_content = ""
+            if ocr_response.pages:
+                for page in ocr_response.pages:
+                    if hasattr(page, 'markdown') and page.markdown:
+                        markdown_content += page.markdown + "\n\n"
+
+            return markdown_content if markdown_content else "No text extracted from document."
+        
+        except Exception as e:
+            print(f"Mistral OCR error: {e}")
+            raise Exception(f"OCR processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)